src/gcp/cloudfunctions.ts

```diff
@@ -5,9 +5,21 @@ import * as api from "../api";
 import { FirebaseError } from "../error";
 import * as logger from "../logger";
 import * as utils from "../utils";
-import { Operation } from "../functionsDeployHelper";
 import { CloudFunctionTrigger } from "../deploy/functions/deploymentPlanner";
 
+interface Operation {
+  name: string;
+  type: string;
+  funcName: string;
+  done: boolean;
+  eventType?: string;
+  trigger?: {
+    eventTrigger?: any;
+    httpsTrigger?: any;
+  };
+  error?: { code: number; message: string };
```

Comment:
I'm surprised there isn't already a type for a Google error. I'm also vaguely surprised that there isn't already an Operation type. E.g. this could roughly be

```
import {Operation} from '../cloud/basicTypes';
import {CloudFunction} from '../cloud/cloudFunctions';

interface CloudFunctionOperation = Operation | CloudFunction;
```